cpp/src/arrow/dataset/scanner.cc

```diff
@@ -32,6 +32,10 @@
 #include "arrow/dataset/dataset.h"
 #include "arrow/dataset/dataset_internal.h"
 #include "arrow/dataset/scanner_internal.h"
+#include "arrow/io/memory.h"
```

Comment:
nit: are these includes necessary here?

Reply:
Scanner.cc does not need any change anymore.